tla-io/src/main/scala/at/forsyte/apalache/io/quint/QuintIR.scala

```diff
@@ -5,6 +5,29 @@ package at.forsyte.apalache.io.quint
 //
 // See https://com-lihaoyi.github.io/upickle/#Builtins for documentation on
 // upickle semi-automatic JSON serialization docs.
+//
+//
+//////////////////////////
+// >> DEBUGGING TIPS << //
+//////////////////////////
+//
+// When serialization fails, it will give an error of the form
+//
+//    expected <JSON value> got <diff JSON value> at index <N>
+//
+// E.g.,
+//
+//    expected sequence got dictionary at index 525
+//
+// The _index_ is the character count of the JSON data. To debug,
+// first get quint's JSON representation of the spec with
+//
+//    quint typecheck --out spec.qnt.json spec.qnt
```

Comment:
This looks a bit hard to debug. Would not it be easier (for me :)) to save the offending json and print the filename?

Reply:
We don’t have file names
At this point. But file names are printed via the top level error handler when this is activated h the cli. 

Unfortunately upickle just doesn’t offer very nice error messages when using its generated serialization/deserialization :( this probably wouldn’t be a problem if we were using one of the more popular Json libs? If we were manually deserializing everything we could do our own error handling of course, but I don’t think it would be worth the extra code.

I’m not very happy with upickle so far tho, so maybe we can swap it out to get better debugging. The only issue is that upickle was selected for the Apalache IR, and I’m
Not sure we want two competing serialization formats :)

Reply:
I’ll think on it!